File: Hello Algorithm/Hello Algorithm_OC/Hello Algorithm/Hello Algorithm/Test/Person.m
//
//  Person.m
//  Hello Algorithm
//
//  Created by mac on 2021/4/29.
//

#import "Person.h"

@implementation Person

+ (instancetype) personWithName:(NSString *)name {
    return [[self alloc] initWithName:name];
}

- (instancetype)initWithName:(NSString *)name
{
    self = [super init];
    if (self) {
        self.name = name;
    }
    return self;
}

- (void)dealloc {
    NSLog(@"%@ -- dealloc",self.name);
}

- (NSString *)description {
    return self.name;
}

- (BOOL)isEqual:(id)other
{
    if ([other isMemberOfClass:[self class]]) {
        Person *otherPerson = other;
        if ([otherPerson.name isEqualToString:self.name]) {
            return YES;
        }
    }
    return NO;
}



@end

File: Hello Algorithm/Hello Algorithm_Swift/Hello Algorithm/Hello Algorithm/Utils&Test/Person.swift
//
//  Person.swift
//  Hello Algorithm
//
//  Created by mac on 2021/7/20.
//

import Foundation

class Person: Equatable,CustomStringConvertible {
    var name: String
    var age: Int
    
    init(name: String,age: Int) {
        self.name = name
        self.age = age
    }
    
    static func == (lhs: Person, rhs: Person) -> Bool {
        if lhs.name == rhs.name && lhs.age == rhs.age  {
            return true
        }
        return false
    }
    
    var description: String {
        return "Person(\(name))"
    }
    
    
    
    
}
